### PR TITLE
ELEMENTS-709: Handle obsolete responses on nuxeo-resource and nuxeo-o…

### PR DESCRIPTION
**Pull request by RSalem07**
…peration

### CODE REVIEW

File: core/nuxeo-operation.js [hunk 1]
```diff
@@ -75,6 +75,16 @@ import './nuxeo-page-provider.js';
           notify: true,
         },
 
+        /** Indicates if the request behind this operation cannot be cancelable.
+         * By default a request is cancelable, it means if on the same `nuxeo-operation`
+         * we perform two sequential requests: Request A & Request B. The first one will be
+         * aborted and we keep the last one (in our case request B). This is done to avoid
+         * an obsolete responses.
+         * */
+        notCancelable: {
```
Comment (nmpcunha) [inline core/nuxeo-operation.js]: Despite not being perfect, for readability purposes I would suggest to call the property `uncancelable`

Comment (nmpcunha) [reply core/nuxeo-operation.js]: Despite that 👍

File: core/nuxeo-operation.js [hunk 2]
```diff
@@ -75,6 +75,16 @@ import './nuxeo-page-provider.js';
           notify: true,
         },
 
+        /** Indicates if the request behind this operation cannot be cancelable.
```
Comment (Gabez0r) [inline core/nuxeo-operation.js]: Either
```suggestion
        /** Indicates if the request behind this operation cannot be canceled.
```
Or
```suggestion
        /** Indicates if the request behind this operation is not cancelable.
```

File: core/nuxeo-operation.js [hunk 3]
```diff
@@ -75,6 +75,16 @@ import './nuxeo-page-provider.js';
...
+         * By default a request is cancelable, it means if on the same `nuxeo-operation`
```
Comment (Gabez0r) [inline core/nuxeo-operation.js]:
```suggestion
         * By default a request is cancelable, which means that if on the same `nuxeo-operation`
```

File: core/nuxeo-resource.js [hunk 4]
```diff
@@ -74,6 +74,16 @@ import './nuxeo-connection.js';
           notify: true,
         },
 
+        /** Indicates if the request behind this resource cannot be cancelable.
```
Comment (Gabez0r) [inline core/nuxeo-resource.js]: Same changes as above.

File: core/nuxeo-operation.js (hunk 2 above)
Comment (RSalem07) [reply core/nuxeo-operation.js]: :

Comment (RSalem07) [reply core/nuxeo-operation.js]: Go for the first proposition:  /** Indicates if the request behind this operation c**annot be canceled**.

File: core/nuxeo-operation.js [hunk 5]
```diff
@@ -228,6 +238,18 @@ import './nuxeo-page-provider.js';
           options.headers[`enrichers-${type}`] = v;
```
Comment (guirenard) [inline core/nuxeo-operation.js]: not related at all but I see that nuxeo-operaiton is importing nuxeo-page-provider element and I don't see a reason why.
Would you mind cleaning that up using your ticket?

Comment (RSalem07) [reply core/nuxeo-operation.js]: yep, of course ;)

Comment (RSalem07) [reply core/nuxeo-operation.js]: i made some checks before removing this import and i see the code [here](https://github.com/nuxeo/nuxeo-elements/blob/f8e865124142c5305f63b557df7395bd18930209/core/nuxeo-operation.js#L199) there is a check on the `input` if it's type is a PageProvider 
perhaps i miss some thing but i think the import is required here for this check.

Comment (Gabez0r) [reply core/nuxeo-operation.js]: Indeed, it is required there.

Comment (RSalem07) [reply core/nuxeo-operation.js]: 👍

**Pull request merged**